explorer/interpreter/impl_scope.h

```diff
@@ -52,6 +52,13 @@ class ImplScope {
            Nonnull<const Value*> type,
            llvm::ArrayRef<Nonnull<const ImplBinding*>> impl_bindings,
            Nonnull<const Witness*> witness, const TypeChecker& type_checker);
+  // Add a list of impl constraints from a constraint type into scope. Any
+  // references to `.Self` are expected to have already been substituted for
+  // the type implementing the constraint.
+  void Add(llvm::ArrayRef<ConstraintType::ImplConstraint> impls,
```

Comment:
This constraint is present for all the overloads; it's just not likely to come up in the other cases because they're usually not referring to constraints that have a `.Self`. The semantics are the same in all of the overloads.

Reply:
I wonder if that could be made clearer without adding much more boilerplate text?

Reply:
Done.